syfertext/language.py

```diff
@@ -84,55 +70,341 @@ def __init__(
+        # as done by spaCy
+        assert (
+            isinstance(name, str) and len(name) >= 1
+        ), "Argument `name` should be of type `str` with at least one character."
+
+        # [TODO] Add custom error message
+        assert (
+            name not in self.pipe_names
+        ), "Pipeline component name '{}' that you have chosen is already used by another pipeline component.".format(
+            name
+        )
+
+        # Make sure only one of 'before', 'after', 'first' or 'last' is set
+        # [TODO] Add custom error message
+        assert (
+            sum([bool(before), bool(after), bool(first), bool(last)]) < 2
+        ), "Only one among arguments 'before', 'after', 'first' or 'last' should be set."
+
+        # Add the new pipe component to the list of factories
+        self.factories[name] = component
+
+        # Create the pipe template that will be added the pipeline
+        # template
+        pipe_template = dict(remote=remote, name=name)
+
+        # Add the pipe template at the right position
+        if last or not any([before, after, first]):
+            self.pipeline_template.append(pipe_template)
+
+        elif first:
+            # The index 0 is reserved for the tokenizer
+            self.pipeline_template.insert(index=1, element=pipe_template)
+
+        elif before in self.pipe_names:
+            self.pipeline_template.insert(
+                index=self.pipe_names.index(before), element=pipe_template
+            )
+
+        elif after in self.pipe_names:
+            self.pipeline_template.insert(
+                index=self.pipe_names.index(after) + 1, element=pipe_template
+            )
+        else:
+            # [TODO] Raise exception with custom error message
+            assert (
+                False
+            ), "component cannot be added to the pipeline, \
+                please double check argument values of the `add_pipe` method call."
+
+        # Reset the pipeline.
+        # The instance variable that will be affected is:
+        # self.pipeline
+        self._reset_pipeline()
+
+    def remove_pipe(self, name: str) -> Tuple[str, callable]:
+        """Removes the pipeline whose name is 'name'
+
+        Args:
+            name (str): The name of the pipeline component to remove.
+
+        Returns:
+            The removed pipe 
+
+        """
+
+        # [TODO] Add custom error message
+        assert (
+            name in self.pipe_names
+        ), "No pipeline component with the specified name '{}' was found".format(name)
+
+        # Get the index of the pipeline to be removed in the
+        # self.pipeline list
+        pipe_index = self.pipe_names.index(name)
+
+        # Delete the pipe using its index
+        pipe = self.pipeline_template.pop(pipe_index)
+
+        # Parse the pipeline template again
+        # to create the subpipeline templates
+        self._parse_pipeline_template()
+
+        # Reset the pipeline.
+        self._reset_pipeline()
```

Comment:
@AlanAboudib Do we need to call `_parse_pipeline_template` ? I think its being already done in `_reset_pipeline`. So Its being done twice currently.

Reply:
@bicycleman15 Good catch. I have changed `_reset_pipeline()` to include `_parse_pipeline_templte()` and forgot to remove it here. Do not hesitate to create a PR for this